### PR TITLE
Feature/ion accordion

### PR DESCRIPTION
**Pull request by mszyld**
Still ToDo: 

- Is the approach with 4 components (article, preprint, talk, project) ok?

- How to make the functions isArticle, isTalk, isProject available to two modules without having to type them again?

### CODE REVIEW

File: src/app/home/home.page.ts [hunk 1]
```diff
@@ -1,12 +1,282 @@
...
+      this.added_stuff = true;
+    }
+
+  }
+
+  // getDetails(item: Article | Project) {
+  //     return this.isArticle(item) ? [item.description, item.collab, item.published, item.url]: item.description;
+  // }
+
+  isArticle(item: Article | Project | Talk): item is Article {
```
Comment (natron2244) [inline src/app/home/home.page.ts]: You can just make these straight static functions, add them to `DataDefinitions`.

**Review by natron2244 — CHANGES_REQUESTED** [review]

File: src/app/home/DataDefinitions.ts [hunk 2]
```diff
@@ -0,0 +1,42 @@
+// import { Injectable } from '@angular/core';
+
+export interface PersonalItem {
```
Comment (natron2244) [inline src/app/home/DataDefinitions.ts]: I would renaming to interfaces.ts

File: src/app/home/home.page.ts (hunk 1 above)
Comment (mszyld) [reply src/app/home/home.page.ts]: should these functions be part of a class?

Comment (natron2244) [reply src/app/home/home.page.ts]: No they don't have to be, can be imported. They can be static functions in a file.

File: src/app/home/home.page.ts [hunk 3]
```diff
@@ -1,12 +1,282 @@
...
+@Component({
+  selector: 'app-home',
+  templateUrl: 'home.page.html',
+  styleUrls: ['home.page.scss'],
+})
+export class HomePage {
+
+  public name = "Papers and Talks by Projects";
+
+  public articles: Article[] = [
```
Comment (natron2244) [inline src/app/home/home.page.ts]: I would move these data objects into there own data files.

**Review by natron2244 — CHANGES_REQUESTED** [review]

File: src/app/home/home.page.ts [hunk 4]
```diff
@@ -1,12 +1,282 @@
...
-})
-export class HomePage {
-
-  constructor() {}
-
-}
+import { Component } from '@angular/core';
+
+import {PersonalItem, PersonalWorkItem, Project, Article, Talk} from './DataDefinitions' //, isArticle,  DataFunctions
+//import { isArticle, isTalk, isProject } from './DataDefinitions';
```
Comment (natron2244) [inline src/app/home/home.page.ts]: remove unused comments

File: src/app/home/home.page.ts [hunk 5]
```diff
@@ -1,12 +1,282 @@
...
+  public added_stuff = false;
+
+  constructor( ) {
+  }
+
+  segmentHasChanged(value) {
+    this.currentList = value.detail.value;
+    
+    // The first time value changes, create the list of items of each project:
+    if (this.added_stuff == false) {
```
Comment (natron2244) [inline src/app/home/home.page.ts:34]: I would not do this in this class, rather make a helper service to load the data.

Comment (natron2244) [reply src/app/home/home.page.ts:34]: https://angular.io/guide/architecture-services

File: src/app/home/home.page.html [hunk 6]
```diff
@@ -1,20 +1,58 @@
...
+    
+    <div *ngIf="isArticle(currentList[0])">
+      <h2>Articles</h2> 
+      <div *ngFor="let item of currentList">
+        <app-article-item *ngIf="item.published" [value]="item">  </app-article-item>  
+      </div>
+
+      <h2>Preprints</h2> 
+      <div *ngFor="let item of currentList">
+        <app-preprint-item *ngIf="item.published==false" [value]="item">  </app-preprint-item>
```
Comment (natron2244) [inline src/app/home/home.page.html]: should be ===

**Pull request merged**